controllers/component_build_controller_pac.go

```diff
@@ -684,12 +730,24 @@ func (r *ComponentBuildReconciler) ensurePaCRepository(ctx context.Context, comp
 		return err
 	}
 	if repository != nil {
+		// Kind of migration. When reusing Repository CR for multi-components
+		// within single source repository, the custom parameters are added to
+		// the existing Repository CR.
```

Comment:
What do you mean by "migration"?

Reply:
There are existing Repository CRs inside users' workspace that need to be updated with corresponding components. For non-multi-components case, the Repository CRs have to be updated separately I think.